src/components/Map/Map.tsx

```diff
@@ -0,0 +1,119 @@
+import UrlIcon from 'leaflet/dist/images/marker-icon.png';
+import RetinaUrlIcon from 'leaflet/dist/images/marker-icon-2x.png';
+import ShadowUrlIcon from 'leaflet/dist/images/marker-shadow.png';
+import { icon } from 'leaflet';
+import {
+  AttributionControl,
+  MapContainer,
+  Marker,
+  TileLayer,
+  useMapEvents,
+} from 'react-leaflet';
+import React from 'react';
+
+import classes from './Map.module.css';
+import 'leaflet/dist/leaflet.css';
+
+// Default is center of Norway
+const DefaultCenter: [number, number] = [64.888996, 12.8186054];
+const DefaultZoom = 4;
+
+// Default map layers from Kartverket
+const DefaultMapLayers = [
+  {
+    url: 'https://opencache.statkart.no/gatekeeper/gk/gk.open_gmaps?layers=europa_forenklet&zoom={z}&x={x}&y={y}',
+    attribution: 'Data © <a href="http://www.kartverket.no/">Kartverket</a>',
+  },
+  {
+    url: 'https://opencache.statkart.no/gatekeeper/gk/gk.open_gmaps?layers=norgeskart_bakgrunn2&zoom={z}&x={x}&y={y}',
+    attribution: 'Data © <a href="http://www.kartverket.no/">Kartverket</a>',
+  },
+];
+
+export interface MapProps {
+  readOnly: boolean;
+  layers?: {
```

Comment:
You could define a `MapLayer`-type and maybe simplify the properties so they are easier to understand?
If not you should add a document to Storybook describing the structure.
For example the anatomy of the url prop. What does {s}, {x}, {y}, {z} mean and are there others etc.?
The main idea is that Leaflet's api-design should not be the main interface, if we at some point decide to change map libraries.
Also is this optional?